cmd/serverless/main.go

```diff
@@ -297,7 +297,17 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	// start the proxy if needed
 	// as this feature is experimental, this should be the only entrypoint (easier to remove)
-	_ = proxy.Start("127.0.0.1:9000", "127.0.0.1:9001", &invocationlifecycle.ProxyProcessor{})
+	if serverlessDaemon.MetricAgent != nil {
+		_ = proxy.Start(
+			"127.0.0.1:9000",
+			"127.0.0.1:9001",
+			&invocationlifecycle.ProxyProcessor{
+				ExtraTags:     serverlessDaemon.ExtraTags,
+				MetricChannel: serverlessDaemon.MetricAgent.GetMetricChannel(),
+			})
+	} else {
+		log.Error("The metric agent has not been set. Unable to start the proxy")
```

Comment:
Is it possible that the metric agent would be `nil` here? If so, and we don't start the proxy, will the extension work correctly, given that we have already run the boostrap script? @maxday may have some more insight here.

Reply:
correct, this won't work if the bootstrap has already set the env variable, maybe we should just panic()